hyperspy/tests/signal/test_tools.py

```diff
@@ -51,6 +51,16 @@ def test_signal_iterator():
         for i, signal in enumerate(s):
             assert i == signal.data[0]
 
+def test_squeeze():
+    s = signals.Signal2D(np.random.random((2,1,1,6,7,8)))
+    assert s.squeeze().axes_manager.shape == (6, 2, 8, 7)
+    assert s.squeeze().data.shape == (2, 6, 7, 8)
+    s = signals.BaseSignal(np.random.random((2,1,1,6,8,7,1,1)))
```

Comment:
Wondering if this is worth adding another order of axis size which should give the same output, so that we can easily use `pytest.mark.parametrize` marker. For example, the following case should give the same output:
```python
s = hs.signals.BaseSignal(np.random.random((2, 1, 1, 6, 8, 7, 1, 1)))
....
s = hs.signals.BaseSignal(np.random.random((1, 1, 2, 6, 8, 1, 1, 7)))
```

Reply:
Why not, I added three versions each for the two tested shapes